cosmwasm/contracts/ibc-translator/src/execute.rs

```diff
@@ -0,0 +1,298 @@
+                    amount: bridging_coin.amount,
+                },
+                recipient_chain: chain,
+                recipient,
+                fee: Uint128::from(0u128),
+                payload,
+                nonce,
+            }
+        }
+    };
+    let initiate_transfer_msg = to_binary(&token_bridge_transfer)
+        .context("could not serialize token bridge initiate_transfer msg")?;
+    response = response.add_message(CosmosMsg::Wasm(WasmMsg::Execute {
+        contract_addr: token_bridge_contract,
+        msg: initiate_transfer_msg,
+        funds: vec![],
+    }));
+
+    Ok(response)
+}
+
+pub fn parse_bank_token_factory_contract(
+    deps: DepsMut,
+    env: Env,
+    coin: Coin,
+) -> Result<String, anyhow::Error> {
+    // extract the contract address from the denom of the token that was sent to us
+    // if the token is not a factory token created by this contract, return error
+    let parsed_denom = coin.denom.split('/').collect::<Vec<_>>();
+    ensure!(
+        parsed_denom.len() == 3
+            && parsed_denom[0] == "factory"
+            && parsed_denom[1] == env.contract.address,
+        "coin is not from the token factory"
+    );
+
+    // decode subdenom from base64 => encode as cosmos addr to get contract addr
+    let cw20_contract_addr = contract_addr_from_base58(deps.as_ref(), parsed_denom[2])?;
+
+    // validate that the contract does indeed match the stored denom we have for it
+    let stored_denom = CW_DENOMS
+        .load(deps.storage, cw20_contract_addr.clone())
+        .context(
+            "a corresponding denom for the extracted contract addr is not contained in storage",
+        )?;
+    ensure!(
+        stored_denom == coin.denom,
+        "the stored denom for the contract does not match the actual coin denom"
+    );
+
+    Ok(cw20_contract_addr)
+}
+
+pub fn contract_addr_from_base58(deps: Deps, subdenom: &str) -> Result<String, anyhow::Error> {
+    let decoded_addr = bs58::decode(subdenom)
+        .into_vec()
+        .context(format!("failed to decode base58 subdenom {subdenom}"))?;
+    let canonical_addr = Binary::from(decoded_addr);
+    deps.api
+        .addr_humanize(&canonical_addr.into())
+        .map(|a| a.to_string())
+        .context(format!("failed to humanize cosmos address {subdenom}"))
+}
+
+pub fn submit_update_chain_to_channel_map(
+    deps: DepsMut,
+    env: Env,
+    _info: MessageInfo,
+    vaa: Binary,
+) -> Result<Response<TokenFactoryMsg>, anyhow::Error> {
+    // get the token bridge contract address from storage
+    let wormhole_contract = WORMHOLE_CONTRACT
+        .load(deps.storage)
+        .context("could not load token bridge contract address")?;
+
+    let vaa: ParsedVAA = deps.querier.query(&QueryRequest::Wasm(WasmQuery::Smart {
+        contract_addr: wormhole_contract,
+        msg: to_binary(&WormholeQueryMsg::VerifyVAA {
+            vaa,
+            block_time: env.block.time.seconds(),
+        })?,
+    }))?;
```

Comment:
Since we have the `wormhole-bindings` package, which lets us call into the wormhole core module, let's call into that instead of making the contract query to the wormhole contract. See an example here -- 

https://github.com/wormhole-foundation/wormhole/blob/main/cosmwasm/contracts/wormchain-ibc-receiver/src/contract.rs#L58-L73